synapse/lib/storm.py

```diff
@@ -494,6 +494,22 @@ def canPropName(self, name):
                 continue
             self.vars[name] = valu
 
+    async def propBackVars(self, runt):
+        '''
+        From a called runt (passed in by parameter) propagate the vars that are not functions
+        and do not conflict with the calling runt.  If there is a conflict, raise StormRuntimeError.
+        '''
+        for name, valu in runt.vars.items():
+            if valu is s_common.novalu:
+                continue
+            if not self.canPropName(name):
+                # don't override our parent's version of a function
+                continue
+            if name in self.runtvars:
```

Comment:
This sounds sane at first pass but is not okay, since it allows an unknown opaque block of storm to declare variables which are not okay for the outer scope to use.